s2s/time.py

```diff
@@ -414,41 +412,108 @@ def __str__(self):
             pass
         else:
-            raise ValueError("Of start/end/periods, specify exactly 2")
+            raise ValueError("Specify either start or periods, not both.")
+
         raise NotImplementedError
 
     def get_lagged_indices(self, lag=1):  # noqa
         """Return indices shifted backward by given lag."""
         raise NotImplementedError
 
-    def get_train_indices(self, strategy, params):  # noqa
-        """Return indices for training data indices using given strategy."""
-        raise NotImplementedError
+    @property
+    def traintest(self) -> pd.DataFrame:
+        """Shorthand for getting both train and test indices.
 
-    def get_test_indices(self, strategy, params):  # noqa
-        """Return indices for test data indices using given strategy."""
-        raise NotImplementedError
+        The user will get a flat intervals list similar to the output of `map_years`
+        or `map_data`, but with extra columns containing `train` and `test` labels.
 
-    def get_train_test_indices(self, strategy, params):  # noqa
-        """Shorthand for getting both train and test indices."""
-        # train = self.get_train_indices()
-        # test = self.get_test_indices()
-        raise NotImplementedError
+        Returns:
+            Pandas DataFrame with columns specifying whether the interval is
+                part of the train or test datasets.
+
+        Example:
+
+            >>> import s2s.time
+            >>> calendar = s2s.time.AdventCalendar(anchor_date=(10, 15), freq='180d')
+            >>> calendar.map_years(2020, 2021).flat
+            anchor_year  i_interval
+            2021         0             (2021-04-18, 2021-10-15]
+                         1             (2020-10-20, 2021-04-18]
+            2020         0             (2020-04-18, 2020-10-15]
+                         1             (2019-10-21, 2020-04-18]
+            dtype: interval
+
+            >>> calendar.set_traintest_method("kfold", n_splits = 2)
+            >>> traintest_group = calendar.traintest
+            >>> traintest_group
+                                                              0                         1
+            anchor_year fold_0 fold_1
+            2021        test   train   (2021-04-18, 2021-10-15]  (2020-10-20, 2021-04-18]
+            2020        train  test    (2020-04-18, 2020-10-15]  (2019-10-21, 2020-04-18]
+        """
+        if self._traintest is None:
+            raise RuntimeError("The train/test splitting method has not been set yet.")
+        df_combined = self._intervals.join(self._traintest)
+        new_index_cols = [self._intervals.index.name] + list(self._traintest.columns)
+        return df_combined.reset_index().set_index(new_index_cols)  # .stack()
+
+    def set_traintest_method(
+        self, method: str, overwrite: bool = False, **method_kwargs: Optional[dict]
+    ):
+        """
+        Configure the train/test splitting strategy for this calendar instance.
+
```

Comment:
Can we add a hyperlink to the documentation for s2s.traintest that will render on readthedocs?

Reply:
https://ai4s2s.readthedocs.io/en/latest/autoapi/s2spy/time/index.html
This link will be added to the docstring, to tell the user the where to find a list of train/test splitting methods supported by this function.